### PR TITLE
feat(carto): Allow configuring MAX_GET_LENGTH

### PR DESCRIPTION
**Pull request by donmccurdy**
Currently MAX_GET_LENGTH, used internally in the CARTO module, is hard-coded to 8192. With this change the default limit is lowered, and is exposed for user configuration to allow future changes in userland.

Example:

```javascript
const data = vectorTableSource({
  ...
  maxLengthURL: 5000
});
```

### CODE REVIEW

File: modules/carto/src/api/common.ts [hunk 1]
```diff
@@ -1,4 +1,22 @@
...
-export const MAX_GET_LENGTH = 8192;
+
+// Fastly default limit is 8192; leave some padding.
+export let MAX_GET_LENGTH = 7000;
+
+/**
+ * Returns maximum URL character length. Above this limit, requests use POST.
+ * Used to avoid browser and CDN limits.
+ */
+export function getRequestURLLimit(): number {
```
Comment (felixpalmer) [inline modules/carto/src/api/common.ts:12]: I think it is cleaner to add `maxRequestLength` to `SourceOptionalOptions` and connect up in `fetchMap`, `baseSource` & `query`

**Review by felixpalmer — CHANGES_REQUESTED** [review]

File: modules/carto/src/api/common.ts (hunk 1 above)
Comment (donmccurdy) [reply modules/carto/src/api/common.ts]: Done! ✅

Comment (felixpalmer) [reply modules/carto/src/api/common.ts]: Could you also update the docs?

Comment (donmccurdy) [reply modules/carto/src/api/common.ts]: Added to `data-sources.md`. I'd prefer not exposing the new parameter in pydeck-carto at this time, and haven't made changes there.

**Review by felixpalmer — APPROVED** [review]

**Pull request merged**